lesson_02/index.ko.md

```diff
@@ -0,0 +1,181 @@
+# FFmpeg 어셈블리어 강의 2강
+
+첫 번째 어셈블리어 함수를 작성해 보았습니다. 이번에는 분기(branch)와 반복문(loop)을 소개하겠습니다.
+
+먼저 레이블과 점프(jump)의 개념을 알아야 합니다. 아래의 예제에서 `jmp` 명령어는 코드 실행을 `.loop:` 뒤로 이동시킵니다. `.loop:`는 **레이블**이라 부르며, 레이블 앞에 붙은 점(`.`)은 이것이 로컬 레이블(local label)임을 나타냅니다. 로컬 레이블을 사용하면 여러 함수에서 동일한 레이블을 재사용할 수 있습니다.
+
+아래는 무한 루프의 예제입니다. 이후 이를 좀 더 현실적인 예시로 확장할 것입니다.
+
+```assembly
+mov  r0q, 3
+.loop:
+    dec  r0q
+    jmp .loop
+```
+
+더욱 현실적인 반복문을 만들기 전에 FLAGS 레지스터를 알아보겠습니다. FLAGS의 세부 동작에 대해서는 깊게 다루지 않겠습니다(앞서 설명했듯이 GPR 연산은 주로 보조 역할이기 때문입니다). 다만, 산술 연산이나 시프트처럼 스칼라 데이터에 대해 `mov` 이외의 대부분의 명령을 실행하면 **Zero-Flag**, **Sign-Flag**, **Overflow-Flag**와 같은 여러 플래그가 설정됩니다.
+
+다음은 루프 카운터가 0까지 감소하도록 하고, `jg`(0보다 크면 점프)가 반복 조건이 되는 예시입니다. `dec r0q` 명령은 실행 후 `r0q` 값에 따라 플래그들을 설정하며, 이 플래그를 바탕으로 점프할 수 있습니다.
+
+```assembly
+mov  r0q, 3
+.loop:
+    ; ...
+    dec  r0q
+    jg  .loop ; 0보다 크면 점프
+```
+
+이는 다음 C 코드와 동일합니다.
+
+```c
+int i = 3;
+do
+{
+   // ...
+   i--;
+} while(i > 0)
+```
+
+이 코드는 약간 부자연스럽습니다. 일반적인 C코드의 반복문은 다음과 같이 작성합니다.
+
+```c
+int i;
+for(i = 0; i < 3; i++) {
+    // ...
+}
+```
+
+이 `for` 루프와 동일하게 대응시키는 방법은 없지만, 대략 다음 어셈블리 코드와 비슷합니다.
+
+```assembly
+xor r0q, r0q
+.loop:
+    ; ...
+    inc r0q
+    cmp r0q, 3
+    jl  .loop ; jump if (r0q - 3) < 0, i.e (r0q < 3)
+```
+
+이 코드 조각에서 주목해야 할 점이 몇 가지 있습니다.
+
+첫째, `xor r0q, r0q`는 레지스터를 0으로 만드는 흔한 방법입니다. 일부 시스템에서는 `mov r0q, 0`보다 더 빠를 수 있는데, 간단히 설명하면 실제 메모리 로드 동작이 일어나지 않기 때문입니다. 이 방식은 SIMD 레지스터에서도 사용할 수 있으며, 예를 들어 `pxor m0, m0`[^1]는 전체 레지스터를 0으로 초기화합니다.
+
+둘째, `cmp`의 사용입니다. `cmp`는 첫 번째 피연산자에서 두 번째 피연산자를 뺀 값을 어디에도 저장하지 않고, 그 결과에 따라 FLAGS를 설정합니다. 주석의 내용처럼 점프 명령과 함께 읽어야 하며, `jl`(jump if less than zero)는 `r0q < 3`일 때 점프하게 됩니다.
```

Comment:
> 주석의 내용처럼 점프 명령과 함께 읽어야 하며

Original: ..., but as per the comment, it can be read together with the jump, ...

I suggest "주석의 내용처럼 점프 명령과 같이 조합되면". as we still get the meaning where cmp can be combined with jump instructions.